src/main/java/com/amazon/opendistroforelasticsearch/security/transport/OpenDistroSecurityInterceptor.java

```diff
@@ -117,6 +118,7 @@ public OpenDistroSecurityInterceptor(final Settings settings,
 
         final Map<String, String> origHeaders0 = getThreadContext().getHeaders();
         final User user0 = getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+        final String injectedUserString = getThreadContext().getTransient(ConfigConstants.OPENDISTRO_SECURITY_INJECTED_USER);
```

Comment:
Is there a need to pass both users?

What would user0 be in case of injection?

Reply:
Only one user will be part of the request - Inject user will take precedence if both the user0 and injectedUserString are present.